onnxruntime/core/providers/webgpu/tensor/concat.cc

```diff
@@ -78,13 +78,13 @@ Status ConcatProgram::GenerateShaderCode(ShaderHelper& shader) const {
   AppendCalCulateInputIndexFunction(shader.AdditionalImplementation(), input_count);
   // add implementation of fn assign_output_data
   AppendAssignOutputDataFunction(shader.AdditionalImplementation(), inputs, output);
-
+  const std::string size_in_concat_axis = GetElementAt("uniforms.size_in_concat_axis", "input_index - 1", input_count);
   shader.MainFunctionBody() << shader.GuardAgainstOutOfBoundsWorkgroupSizes("uniforms.output_size")
                             << "  var indices = " << output.OffsetToIndices("global_idx") << ";\n"
                             << "  let indices_axis = " << output.IndicesGet("indices", axis_) << ";\n"
                             << "  let input_index = calculate_input_index(indices_axis);\n"
-                               "  if (input_index != 0u) {\n"
-                            << "     " << output.IndicesSet("indices", axis_, "indices_axis - uniforms.size_in_concat_axis[input_index - 1]") << ";\n"
+                            << "  if (input_index != 0u) {\n"
+                            << "    " << output.IndicesSet("indices", axis_, "indices_axis - " + size_in_concat_axis) << ";\n"
```

Comment:
Can we continue to use shift operator here instead of string concatenation to be consistent with the rest of the code?

Reply:
I find it difficult to design a way that accepts L-shift style string concatenation for function parameter. a lambda style may look wired:

```c++
// old
output.IndicesSet("indices",
                  axis_,
                  "indices_axis - " + size_in_concat_axis)

// new
output.IndicesSet("indices",
                  axis_,
                  [](OstringStream &ss) {
                    ss << "indices_axis - " << size_in_concat_axis;
                  });
```